x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/TransportGetStackTracesAction.java

```diff
@@ -208,42 +232,78 @@ private void searchEventGroupedByStackTrace(
     ) {
         GetStackTracesResponseBuilder responseBuilder = new GetStackTracesResponseBuilder();
         responseBuilder.setSamplingRate(eventsIndex.getSampleRate());
+        responseBuilder.setRequestedDuration(request.getRequestedDuration());
```

Comment:
This needs to be set also in `searchGenericEvents`. Can we refactor this so that we create `GetStackTracesResponseBuilder` in `doExecute` and set common properties there as well?

Reply:
I see that `GetStackTracesResponseBuilder` is now initialized both in `searchProfilingEvents` and `searchGenericEvents`. What I meant originally is that we should aim to set common properties already in `doExecute` before we even call these methods. This reduces the potential for bugs in the future.